BDArmory/Weapons/Missiles/BDModularGuidance.cs

```diff
@@ -1274,10 +1274,10 @@ private void SetRoll()
         {
             var vesselTransform = vessel.transform.position;
 
-            Vector3 gravityVector = FlightGlobals.getGeeForceAtPosition(vesselTransform).normalized;
-            Vector3 rollVessel = -vessel.transform.right.normalized;
+            Vector3 upVector = VectorUtils.GetUpDirection(vesselTransform);
```

Comment:
This could be simplified further to
```csharp
Vector3 up = vessel.up;
Vector3 right = vessel.transform.right;
float currentAngle = Vector3.SignedAngle(right, up, Vector3.Cross(right, up)) - 90f;
```
Also, all the `this.` are superfluous.